NATS Manager Documentation for User (#94)
kyma-project/nats-manager: docs/user/01-manager.md

# NATS Manager

This module ships the NATS Manager.

## Module Lifecycle

### Starting NATS Manager

> Upon starting the NATS Manager, the controller (following the [Kubebuilder concept](https://book.kubebuilder.io/architecture.html))

Reviewer: ```suggestion
Upon starting the NATS Manager, the controller (following the [Kubebuilder concept](https://book.kubebuilder.io/architecture.html)) creates, watches and reconciles the following resources:
```

> creates, watches and reconciles the following resources:

Reviewer: ```suggestion
```

   
   - ConfigMap (cm)
   - Secret (sc)
   - Service (sv)

>    - Stateful Set (sfs)

Reviewer 2: ```suggestion
   - Stateful Set (sts)
```

>    - DestinationRule (dr)

Reviewer 2: ```suggestion
   - DestinationRule (dr, [Istio](https://istio.io))
```

   ```mermaid
    graph LR

>      A(Start NATS manager) -->|Controller| D(Creates, watches & reconciles resources: cm, sc, sv, sfs, dr)

Reviewer 2: ```suggestion
     A(Start NATS manager) -->|Controller| D(Creates, watches & reconciles resources: cm, sc, sv, sts, dr)
```

   ```

### Reacting to NATS CR changes

The NATS Manager reacts to changes of the NATS CR to adapt the resources mentioned above to the desired state.
For details how to configure NATS using the CR, visit the [Configuration documentation](./02-configuration.md).

   ```mermaid
   graph LR
     E(NATS CR changes)-->F(Reconciliation triggered)-->|Controller|G(Resources are adapted to reflect the changes)
   ```

### Reacting to resource changes

When resources are changed or deleted, the controller reacts by restoring the defaults according to the NATS CR.

> Thus, changes cannot be made directly to the resources, but the NATS CR needs to be edited.

Reviewer 2: ```suggestion
Thus, changes cannot be made directly to the resources, but via editing the NATS CR.
```
